Add Settings Sync Docs (#1914)
gitpod-io/website: gitpod/docs/ides-and-editors/settings-sync.md

---
section: ides-and-editors
title: VS Code Settings Sync
---

<script context="module">
  export const prerender = true;
</script>

# {title}

> VS Code is so beloved hugely because of its endless possibilities for customization. A problem arises when there are multiple machines or VS Code instances you'd like to have with the same extensions, themes and settings.

Reviewer: I think it makes sense here to reference that each workspace start is a new machine basically. Maybe link to https://www.gitpod.io/docs/workspaces for the context about ephemeral nature of Gitpod

The solution for exactly this and a bunch more is called Settings Sync. It saves all your preferences to the cloud so that you simply log in to the same account on all of your instances and have all your settings in sync.

## How does it work?

When Settings Sync is running, it polls the backend for changes on an interval and at the same time makes sure that after any change to your preferences, these changes are pushed back. After the initial (first) sync, only the changed items are pushed to the backend server to save bandwidth and make the syncronization nearly instant.

The following items are synced when you use Settings Sync:

> - Settings - your VS Code preferences [[docs](https://code.visualstudio.com/docs/getstarted/settings)]

Reviewer: You are forgot about extensions! They are actually most tricky. Because users don't realize that they install extensions for all workspaces by default. And tehy should select `install (no sync)` to install only for given workspace. Maybe it should be mentioned somewhere.

Author: Fixed now.

---
section: ides-and-editors
title: VS Code Settings Sync
---

<script context="module">
  export const prerender = true;
</script>

# {title}

VS Code is so beloved hugely because of its endless possibilities for customization. A problem arises when there are multiple machines or VS Code instances you'd like to have with the same extensions, themes and settings.

> Every Gitpod workspace is basically its own machine, meaning that if you open a new workspace or restart an existing one, your unsynced preferences will not carry over ([why?](https://www.gitpod.io/docs/workspaces)). The solution for exactly this and a bunch more is called Settings Sync. It saves all your preferences to the cloud so that you simply log in to the same account on all of your instances and have all your settings in sync.

Reviewer: [Settings Sync](https://code.visualstudio.com/docs/editor/settings-sync) allows you to share VS Code settings and extensions between Gitpod and other instances of VS Code. Each start of [Gitpod workspace](https://www.gitpod.io/docs/workspaces) is its own instance of VS Code.

## How does it work?

> When Settings Sync is running, it polls the backend for changes on an interval and at the same time makes sure that after any change to your preferences, these changes are pushed back. After the initial (first) sync, only the changed items are pushed to the backend server to save bandwidth and make the syncronization nearly instant.

Reviewer 2: ```suggestion
Once enabled enable Settings Sync it polls the backend for changes on an interval and at the same time makes sure that after any change to your preferences, these changes are pushed back. After the initial (first) sync, only the changed items are pushed to the backend server to save bandwidth and make the syncronization nearly instant.
```

Just a suggestion, feel free to drop it - but I started wondering: "What does it mean `running`? Is this something I can opt-in or it just runs as part of Gitpod?

Reviewer 2: `syncronization` -> `synchronization`

Author: Thanks for pointing this out, saying that Settings Sync is "running" can be a bit misleading users, as it is just something they enable and don't have to check that it is "running". Settings Sync is a part of VS Code's source code and on Gitpod we just **enable** it automatically.

- they are installed manually from a `vsix` file

## On the web

### On Gitpod

If you are using Gitpod with VS Code Web, Settings Sync should already work out of the box with no additional configuration needed. You can verify this by opening the Account menu in VS Code and checking that the last line says `Settings Sync is On`.

### On Codespaces, vscode.dev and others

We don't have support for these environments, yet, but are actively looking into it.

## Locally

By default, your local VS Code installation is _most likely_ equipped with a pre-configured setup to sync all of your IDE-related settings via a GitHub or a Microsoft account. This means that your preferences are synced to Microsoft's servers and that means Gitpod has no access to them.

You can change this fact and sync all of your preferences to Gitpod instead. You will not lose any data by doing this, but if you still want to sync your settings across machines we recommend you to do the same on all of them as well.

1. Make sure the [Gitpod extension](https://marketplace.visualstudio.com/items?itemName=gitpod.gitpod-desktop) is installed and enabled.

> 2. Hit <kbd>F1</kbd> to open the Command Pallete and run `Gitpod: Turn on Settings Sync`.

Reviewer 3: I renamed it to `Settings Sync: Enable signing in with Gitpod`
```suggestion
2. Hit <kbd>F1</kbd> to open the Command Pallete and run `Settings Sync: Enable signing in with Gitpod`.
```

Author: Coolsies, thanks!

> 2. Hit <kbd>F1</kbd> to open the Command Pallete and run `Settings Sync: Enable signing in with Gitpod`.

Reviewer 2: ```suggestion
2. Hit <kbd>F1</kbd> to open the Command Palette and run `Settings Sync: Enable signing in with Gitpod`.
```
